daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/SBuiltin.scala

```diff
@@ -1555,90 +1544,88 @@ private[lf] object SBuiltin {
+      val version = machine.tmplId2TxVersion(templateId)
+      val cachedKey =
+        extractKey(NameOf.qualifiedNameOfCurrentFunc, version, templateId, keyValue)
+      if (cachedKey.maintainers.isEmpty) {
+        Control.Error(
+          IE.FetchEmptyContractKeyMaintainers(
+            cachedKey.templateId,
+            cachedKey.lfValue,
           )
-        } else {
-          val keyOpt = SOptional(Some(keyValue))
-          val gkey = cachedKey.globalKey
-          val optTargetTemplateId: Option[TypeConName] = None // no upgrading
-          machine.ptx.contractState.resolveKey(gkey) match {
-            case Right((keyMapping, next)) =>
+        )
+      } else {
+        val keyOpt = SOptional(Some(keyValue))
+        val gkey = cachedKey.globalKey
+        val optTargetTemplateId: Option[TypeConName] = None // no upgrading
+        machine.ptx.contractState.resolveKey(gkey) match {
+          case Right((keyMapping, next)) =>
+            machine.ptx = machine.ptx.copy(contractState = next)
+            keyMapping match {
+              case ContractStateMachine.KeyActive(coid) =>
+                fetchContract(machine, templateId, optTargetTemplateId, coid, keyOpt) {
+                  templateArg =>
+                    getContractInfo(machine, coid, templateId, templateArg, keyOpt) { contract =>
+                      machine.checkKeyVisibility(gkey, coid, operation.handleKeyFound, contract)
+                    }
+                }
+
+              case ContractStateMachine.KeyInactive =>
+                operation.handleKnownInputKey(gkey, keyMapping)
+            }
+
+          case Left(handle) =>
+            def continue: Option[V.ContractId] => (Control[Question.Update], Boolean) = { result =>
+              val (keyMapping, next) = handle(result)
               machine.ptx = machine.ptx.copy(contractState = next)
               keyMapping match {
                 case ContractStateMachine.KeyActive(coid) =>
-                  fetchContract(machine, templateId, optTargetTemplateId, coid, keyOpt) {
-                    templateArg =>
-                      getContractInfo(machine, coid, templateId, templateArg, keyOpt) { contract =>
-                        machine.checkKeyVisibility(gkey, coid, operation.handleKeyFound, contract)
-                      }
-                  }
-
-                case ContractStateMachine.KeyInactive =>
-                  operation.handleKnownInputKey(gkey, keyMapping)
-              }
-
-            case Left(handle) =>
-              def continue: Option[V.ContractId] => (Control[Question.Update], Boolean) = {
-                result =>
-                  val (keyMapping, next) = handle(result)
-                  machine.ptx = machine.ptx.copy(contractState = next)
-                  keyMapping match {
-                    case ContractStateMachine.KeyActive(coid) =>
-                      val c =
-                        machine.asUpdateMachine(productPrefix) { machine =>
```

Comment:
no completly sure why we needed this one

Reply:
:shrug: